dapps/profile.vue/src/components/profile/company/contact/contact.ts

```diff
@@ -0,0 +1,130 @@
+*/
+
+// vue imports
+import Vue from 'vue';
+import Component, { mixins } from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+
+// evan.network imports
+import { EvanComponent, EvanForm, EvanFormControl } from '@evan.network/ui-vue-core';
+import * as bcc from '@evan.network/api-blockchain-core';
+import * as dappBrowser from '@evan.network/ui-dapp-browser';
+
+// internal
+import * as dispatchers from '../../../../dispatchers/registry';
+import ProfileMigrationLibrary from '../../../../lib/profileMigration';
+
+interface ContactFormInterface extends EvanForm {
+  city: EvanFormControl;
+  homepage: EvanFormControl;
+  postalCode: EvanFormControl;
+  streetAndNumber: EvanFormControl;
+}
+
+@Component({ })
+export default class CompanyRegistrationForm extends mixins(EvanComponent) {
+  /**
+   * Address for that the data should be loaded
+   */
+  @Prop() address;
+
+  /**
+   * Evan form instance for registration data.
+   */
+  contactForm: ContactFormInterface = null;
+
+  /**
+   * Watch for dispatcher updates
+   */
+  listeners: Array<any> = [ ];
+
+  /**
+   * Load the mail details
+   */
+  async created() {
+    // watch for save updates
+    this.listeners.push(dispatchers.updateProfileDispatcher.watch(($event: any) => {
+      if ($event.detail.status === 'finished' || $event.detail.status === 'deleted') {
+        this.loadProfileData();
+      }
+    }));
+
+    // load profile data
+    await this.loadProfileData();
+  }
+
+  /**
+   * Clear dispatcher listeners
+   */
+  beforeDestroy() {
+    this.listeners.forEach(listener => listener());
+  }
+
+  /**
+   * Load the mail details
+   */
+  async loadProfileData() {
+    const runtime = (<any>this).getRuntime();
+    let contactData = await ProfileMigrationLibrary.loadProfileData(runtime, 'contact');
+
+    // setup registration form
+    this.contactForm = (<ContactFormInterface>new EvanForm(this, {
+      streetAndNumber: {
+        value: contactData.streetAndNumber || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+      postalCode: {
+        value: contactData.postalCode || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return !!this.value.match(/^\d{5}$/);
+        },
+      },
+      city: {
+        value: contactData.city || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return this.value.length !== 0;
+        },
+      },
+      homepage: {
+        value: contactData.homepage || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return !!this.value.match(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/);
```

Comment:
thats a regex! is it intentional, that it would also matches e-mail addresses?

Reply:
Ahh good tested. I added another one ;)